java/parser.js

```diff
@@ -1,171 +1,206 @@
     }
 
     function layoutRegions(arr) {
-        return arr.reduce((acc, val) => `${acc}.layout(${regionParameter(val)})`, '')
+        return arr.reduce((acc, val) => `${acc}.layout(${regionParameter(val)})`, '');
     }
 
     function floatingRegions(arr) {
-        return arr.reduce((acc, val) => `${acc}.floating(${floating(val)})`, ``)
+        return arr.reduce((acc, val) => `${acc}.floating(${floating(val)})`, ``);
     }
 
     function floating(floating) {
-        let string
-        string = regionParameter(floating.region)
-        string += `, ${floating.maxUpOffset}, ${floating.maxDownOffset}, ${floating.maxLeftOffset}, ${floating.maxRightOffset}`
-        return string
+        let string;
+        string = regionParameter(floating.region);
+        string += `, ${floating.maxUpOffset}, ${floating.maxDownOffset}, ${floating.maxLeftOffset}, ${floating.maxRightOffset}`;
+        return string;
     }
 
     function accessibilityRegions(arr) {
-        return arr.reduce((acc, val) => `${acc}.accessibility(${accessibility(val)})`, ``)
+        return arr.reduce((acc, val) => `${acc}.accessibility(${accessibility(val)})`, ``);
     }
 
     function accessibility(val) {
-        return `${regionParameter(val.region)}, AccessibilityRegionType.${capitalizeFirstLetter(val.type)}`
+        return `${regionParameter(val.region)}, AccessibilityRegionType.${capitalizeFirstLetter(val.type)}`;
     }
 
     function regionParameter(region) {
-        let string
+        let string;
         switch (typeof region) {
             case 'string':
-                string = `By.cssSelector(${JSON.stringify(region)})`
+                string = `By.cssSelector(${JSON.stringify(region)})`;
                 break;
             case "object":
                 if (region.type) {
-                    string = native ? `getDriver().findElement(${parseObject(region)})` : parseObject(region)
+                    string = native ? `getDriver().findElement(${parseObject(region)})` : parseObject(region);
                 } else {
-                    string = parseObject({value: region, type: 'Region'})
+                    string = parseObject({value: region, type: 'Region'});
                 }
                 break;
             case "function":
-                string = serialize(region)
+                string = serialize(region);
                 break;
             default:
-                throw new Error(`Region parameter of the unimplemented type was used:  ${JSON.stringify(region)}`)
+                throw new Error(`Region parameter of the unimplemented type was used:  ${JSON.stringify(region)}`);
         }
         return string
     }
+
+    function checkOptions(actual, supported) {
+        const actualOptions = Object.keys(actual);
+        actualOptions.forEach(option => {
+            if (!supported.includes(option)) throw new Error(`Emitter need update to support check settings option: ${option}`);
```

Comment:
Codacy has a fix for the issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/applitools/sdk.coverage.tests/pullRequest?prid=8528488)
```suggestion
            if (!supported.includes(option)) {throw new Error(`Emitter need update to support check settings option: ${option}`);}
```